app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  def update_post_counter
+    post.increment!(:posts_counter)
+  end
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
```

Comment:
- [ ] Optional Suggestion: You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private.

```
  # app/models/post.rb

  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```

Note: The same way you can use this in 
`# app/models/comment.rb` with `after_save :update_comments_counter` and also in 
`# app/models/like.rb` with `after_save :update_likes_counter`